diff --git a/ckan/lib/uploader.py b/ckan/lib/uploader.py
index 82e2dcbaf..e44fc5f65 100644
--- a/ckan/lib/uploader.py
+++ b/ckan/lib/uploader.py
@@ -343,7 +343,22 @@ class ResourceUpload(object):
                 try:
                     _copy_file(self.upload_file, output_file, max_size)
                 except logic.ValidationError:
-                    os.remove(tmp_filepath)
+                    try:
+                        os.remove(tmp_filepath)
+                    except OSError:
+                        # Other process (blobfuse) might be still using file
+                        # Sleep for a second and try again 5 times
+                        import time
+                        max_attempts = 5
+                        delay = .5  # delay in seconds
+                        for attempt in range(max_attempts):
+                            time.sleep(delay)
+                            try:
+                                os.remove(tmp_filepath)
+                            except OSError as e:
+                                log.warning(f"Failed to remove temporary file attempt {attempt + 1}: {tmp_filepath}. Error: {e}")
+                    except Exception as e:
+                        log.error(f"Failed to remove temporary file: {tmp_filepath}. Error: {e}")
                     raise
                 finally:
                     self.upload_file.close()
