index 06a1758..08a3ba6 100644
--- a/ckanext/archiver/tasks.py
+++ b/ckanext/archiver/tasks.py
@@ -568,14 +568,14 @@ def archive_resource(context, resource, log, result=None, url_timeout=30):
     except Exception:
         file_name = "resource"
 
-    # move the temp file to the resource's archival directory
-    saved_file = os.path.join(archive_dir, file_name)
-    shutil.move(result['saved_file'], saved_file)
-    log.info('Going to do chmod: %s', saved_file)
     try:
-        os.chmod(saved_file, 0o644)  # allow other users to read it
+        # move the temp file to the resource's archival directory
+        saved_file = os.path.join(archive_dir, file_name)
+        shutil.copy(result['saved_file'], saved_file)
+        log.info('File moved : %s', saved_file)
+        os.remove(result['saved_file'])
     except Exception as e:
-        log.error('chmod failed %s: %s', saved_file, e)
+        log.error("move the temp file to the resource's archival directory failed %s: %s", saved_file, e)
         raise
     log.info('Archived resource as: %s', saved_file)
 

~
~