diff --git a/ckanext/archiver/tasks.py b/ckanext/archiver/tasks.py
index 06a1758..95fa968 100644
--- a/ckanext/archiver/tasks.py
+++ b/ckanext/archiver/tasks.py
@@ -570,7 +570,7 @@ def archive_resource(context, resource, log, result=None, url_timeout=30):
 
     # move the temp file to the resource's archival directory
     saved_file = os.path.join(archive_dir, file_name)
-    shutil.move(result['saved_file'], saved_file)
+    shutil.move(result['saved_file'], saved_file, copy_function=shutil.copy)
     log.info('Going to do chmod: %s', saved_file)
     try:
         os.chmod(saved_file, 0o644)  # allow other users to read it
